test/e2e/specs/specs.go

```diff
@@ -242,27 +246,54 @@ func (t *TestPod) VerifyExecInPodFail(f *framework.Framework, containerName, shE
 
 // execCommandInContainerWithFullOutputWithRetry executes a command in a target pod and retries with gradual back until timeout(10 min) or success.
 func execCommandInContainerWithFullOutputWithRetry(f *framework.Framework, podName, containerName string, cmd ...string) (string, string, error) {
+	return RetryWithBackoffTwoReturnValues(func() (string, string, error) {
+		return e2epod.ExecCommandInContainerWithFullOutput(f, podName, containerName, cmd...)
+	})
+}
 
+// Retry executes a generic operation (op) with exponential backoff.
+// T can be any type (string, a struct, a slice, etc).
+func Retry[T any](op func() (T, error)) (T, error) {
 	backoff := wait.Backoff{
 		Duration: backoffDuration,
 		Factor:   backoffFactor,
 		Cap:      backoffCap,
 		Steps:    backoffSteps,
 		Jitter:   backoffJitter,
 	}
-	var err error
-	var stdout, stderr string
+
+	var result T
+	var lastErr error
 
 	wait.ExponentialBackoff(backoff, func() (bool, error) {
-		stdout, stderr, err = e2epod.ExecCommandInContainerWithFullOutput(f, podName, containerName, cmd...)
-		if err != nil {
-			framework.Logf("Exec command failed with error: %v. Retrying...", err)
+		result, lastErr = op()
+
+		if lastErr != nil {
+			framework.Logf("Operation failed with error: %v. Retrying...", lastErr)
 			return false, nil
 		}
 		return true, nil
 	})
 
-	return stdout, stderr, err
+	if lastErr != nil {
+		framework.Logf("Operation failed after %d steps (total time/cap: %v). Last error: %v", backoff.Steps, backoff.Cap, lastErr)
+	} else {
+		framework.Logf("Operation succeeded on a retry.")
```

Comment:
"Operation succeeded on a retry." is displayed even if the operation succeeds on the first attempt without any retries, so this is a little misleading. 

Please change "Operation succeeded on a retry." -> "Operation succeeded." Thanks!